components/Tabs/Tabs.js

```diff
@@ -1,61 +1,62 @@
 class TabLink {
   constructor(tabElement){
     // assign this.tabElement to the tabElement DOM reference
-    // this.tabElement;
+    this.tabElement = tabElement;
     
     // Get the `data-tab` value from this.tabElement and store it here
-    // this.tabData = ; 
+    this.tabData = this.tabElement.dataset.tab;
     
     // We need to find out if a user clicked 'all' cards or a specific category.  Follow the instructions below to accomplish this task:    
     
-    /* <- Delete this comment block when you work on the if statement
+
     // Check to see if this.tabData is equal to 'all'
-    if(){
+    if(this.tabData === "all"){ //strict equality used to see if it is equal
       // If `all` is true, select all cards regardless of their data attribute values
-      // this.cards = ;
+      this.cards = document.querySelectorAll('.card'); //true statement will show all the cards regardless of data attribute
     } else {
       // else if `all` is false, only select the cards with matching this.tabData values
-      // this.cards = ;
-    }
-    /* <- Delete this comment block when you work on the if statement
+      this.cards = document.querySelectorAll(`.card[data-tab='${this.tabData}']`) 
+    } //false statement will just select the cards that match this.tabData attribute
+
+    
 
      // Map over the newly converted NodeList we just created in our if statement above. Convert each this.cards element into a new instance of the TabCard class. Pass in a card object to the TabCard class. 
-    // this.cards = Array.from(this.cards).map();
+    this.cards = Array.from(this.cards).map(card => new TabCard(card)); //Array.from converts to a new array
 
     // Add a click event that invokes this.selectTab
-    // this.tabElement.addEventListener();
-  }
+    this.tabElement.addEventListener('click', () => {this.selectTab()});
+  } //event listener created for when user clicks on a tab, content shows up associated with tab name
 
   selectTab(){
 
     // Select all elements with the .tab class on them
-    // const tabs = document.querySelectorAll();
+    const tabs = document.querySelectorAll('.tab'); //used querySelectorAll() which references all of the tabs on the page
     
     // Iterate through the NodeList removing the .active-tab class from each element
-    // tabs.forEach()
+    tabs.forEach(tab => tab.classList.remove('active-tab'));
 
     // Select all of the elements with the .card class on them
-    // const cards = ;
+    const cards = document.querySelectorAll('.card');
 
     // Iterate through the NodeList setting the display style each one to 'none'
-    // cards.forEach()
+    cards.forEach(tab => tab.style.display = 'none') //each tab is set to display none for the time being until user clicks on specific tab
```

Comment:
the syntax is still correct, but it's actually each card that will be set to display: none until user clicks on a specific tab. It works just because you were consistent in what you called it.